packages/nimble-components/src/table-column/enum-text/tests/table-column-enum-text.spec.ts

```diff
@@ -0,0 +1,420 @@
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'b' }]);
+        await connect();
+        await waitForUpdatesAsync();
+
+        const mapping = model.col1.firstElementChild as MappingText;
+        mapping.key = 'b';
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('alpha');
+    });
+
+    describe('various string values render as expected', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+        for (const value of wackyStrings) {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `data "${value.name}" renders as "${value.name}"`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    ({ element, connect, disconnect, model } = await setup([
+                        { key: 'a', label: value.name }
+                    ]));
+                    pageObject = new TablePageObject<SimpleTableRecord>(
+                        element
+                    );
+                    await element.setData([{ field1: 'a' }]);
+                    await connect();
+                    await waitForUpdatesAsync();
+
+                    expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                        value.name
+                    );
+                }
+            );
+        }
+    });
+
+    describe('various string values render in group header as expected', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+        for (const value of wackyStrings) {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `data "${value.name}" renders as "${value.name}"`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    ({ element, connect, disconnect, model } = await setup([
+                        { key: 'a', label: value.name }
+                    ]));
+                    pageObject = new TablePageObject<SimpleTableRecord>(
+                        element
+                    );
+                    await element.setData([{ field1: 'a' }]);
+                    await connect();
+                    await waitForUpdatesAsync();
+                    model.col1.groupIndex = 0;
+                    await waitForUpdatesAsync();
+
+                    expect(
+                        pageObject.getRenderedGroupHeaderContent(0)
+                    ).toContain(value.name);
+                }
+            );
+        }
+    });
+
+    it('sets group header text to key value when unmatched (instead of blank)', async () => {
```

Comment:
Sorry if I'm starting a discussion that has already been had and settled on, but it seems really odd to me that the group header would show the key value while the column itself would show blank. To me that sees very inconsistent. What's the rationale for the two different behaviors?

Reply:
Did we discuss at some point the sorting inconsistency when the cell shows blank but not all blanks are ordered together when sorting by that column? I think a similar argument to grouping needing to show the raw value could be made for needing to show the raw value when sorting. I'm not sure if that makes me lean towards showing the raw value in both cases or neither case, but I'm not convinced that grouping is a unique problem that should be solved in a special way.

Reply:
That's a good point. I still feel there is a little more of an argument to do something special in the group header, since there is another column (icon, which hasn't been reviewed yet...) where the group header shows something different than the cell. Leslie wanted the icon _and_ its label to be displayed in the group header whereas the cells just show the icon. On the other hand, it would be exceedingly weird if sorting columns made displayed values change. If the general consensus is that we should change the group headers to be blank, like the cells, I'm okay with that.